compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5587,6 +5599,14 @@ private BType resolveQueryType(SymbolEnv env, BLangExpression selectExp, BType c
                 return new BStreamType(TypeTags.STREAM, selectType, errorType, null);
             } else if (queryExpr.isTable) {
                 actualType = getQueryTableType(queryExpr, selectType);
+            } else if (queryExpr.isMap) {
+                BType mapConstraintType = getMapType(selectType, queryExpr.getSelectClause().expression.pos);
+                if (mapConstraintType == symTable.noType) {
+                    actualType = symTable.semanticError;
+                } else {
+                    actualType = BUnionType.create(null, new BMapType(TypeTags.MAP, mapConstraintType, null),
```

Comment:
Why do resolve query result type to have error type always?

Reply:
According to the [spec](https://ballerina.io/spec/lang/2021R1/#query-expr) , 

```
The execution of a clause may complete early with an error value, in which case this error value is the result of the query, except when constructing a stream.
```

So we always need error type here right?

Reply:
It should not always have an `error` type. We have to check whether error returning expressions are nested within the query expression if we are to include the `error` type in the result type. This is being handled through https://github.com/ballerina-platform/ballerina-lang/issues/36181. Let's consider the error-less path here and handle the error cases when the fix for the above mentioned issue is merged

Reply:
Got it. Will update the code. Thanks.

Reply:
If if change it to error-less path then there is a problem when the user have used `on-conflict` clause. I checked the implementation of the query constructing table also. It also always creates a union type including error type. So shall we keep this implementation as it is and fix it properly once  https://github.com/ballerina-platform/ballerina-lang/issues/36181 this is fixed? WDYT @pcnfernando

Reply:
Sure. Will create an issue.

Reply:
https://github.com/ballerina-platform/ballerina-lang/issues/36695